ecommerce/extensions/payment/processors/hyperpay.py

```diff
@@ -0,0 +1,226 @@
+    """
+    An umbrella exception to catch all errors from HyperPay.
+    """
+    pass  # pylint: disable=unnecessary-pass
+
+
+class HyperPay(BasePaymentProcessor):
+    """
+    HyperPay payment processor.
+
+    For reference, see https://hyperpay.docs.oppwa.com/integration-guide and
+    https://hyperpay.docs.oppwa.com/reference/parameters.
+    """
+
+    NAME = 'hyperpay'
+    PAYMENT_MODE = _('Credit card')
+    PAYMENT_TYPE_DEBIT = 'DB'
+    CHECKOUTS_ENDPOINT = '/v1/checkouts'
+    PAYMENT_WIDGET_JS_PATH = '/v1/paymentWidgets.js'
+    RESULT_CODE_SUCCESSFULLY_CREATED_CHECKOUT = '000.200.100'
+    CART_ITEM_TYPE_DIGITAL = 'DIGITAL'
+    BILLING_ADDRESS_STREET1_MAX_LEN = 95
+    BILLING_ADDRESS_STREET2_MAX_LEN = 100
+    BRANDS = "VISA MASTER"
+    CHECKOUT_TEXT = _("Checkout with credit card")
+
+    def __init__(self, site):
+        super(HyperPay, self).__init__(site)
+        configuration = self.configuration
+        self.access_token = configuration['access_token']
+        self.entity_id = configuration['entity_id']
+        self.return_url = configuration['return_url']
+        self.currency = configuration['currency']
+        self.hyper_pay_api_base_url = configuration.get('hyper_pay_api_base_url', 'https://test.oppwa.com')
+        self.collect_billing_details_client_side = configuration.get('collect_billing_details_client_side', True)
+        self.test_mode = configuration.get('test_mode')
+        self.site = site
+
+    @property
+    def authentication_headers(self):
+        """
+        Return the authentication headers.
+        """
+        return {
+            'Authorization': 'Bearer {}'.format(self.access_token)
+        }
+
+    def _get_basket_data(self, basket):
+        """
+        Return the basket data
+        """
+
+        def get_cart_field(index, name):
+            """
+            Return the cart field name.
+            """
+            return 'cart.items[{}].{}'.format(index, name)
+
+        basket_data = {
+            'amount': format_price(basket.total_incl_tax),
+            'currency': self.currency,
+            'merchantTransactionId': basket.order_number
+        }
+        for index, line in enumerate(basket.all_lines()):
+            cart_item = {
+                get_cart_field(index, 'name'): clean_field_value(line.product.title),
+                get_cart_field(index, 'quantity'): line.quantity,
+                get_cart_field(index, 'type'): self.CART_ITEM_TYPE_DIGITAL,
+                get_cart_field(index, 'sku'): line.stockrecord.partner_sku,
+                get_cart_field(index, 'price'): format_price(line.unit_price_incl_tax),
+                get_cart_field(index, 'currency'): self.currency,
+                get_cart_field(index, 'totalAmount'): format_price(line.line_price_incl_tax_incl_discounts)
+            }
+            basket_data.update(cart_item)
+        return basket_data
+
+    def _get_customer_profile_and_billing_data(self, user, request):
```

Comment:
This function and its invocation will be removed since the billing details can be collected in the payment form itself.

Reply:
Can you remove this method to make the code review easier?